rrrspec-client/lib/rrrspec/client/slave_runner.rb

```diff
@@ -57,9 +59,12 @@ def work
           @timeout = TASKQUEUE_TASK_TIMEOUT
           trial = Trial.create(task, @slave)
 
-          @rspec_runner.reset
+          task_file_ext = File.extname(task.spec_file)
+          runner = RunnerFactory.create(task_file_ext)
```

Comment:
I've made a chain of runner factories, to allow plugin the runners. 
The RSpecRunnerFactory has the runner as a singleton, that way the bug is gone. I've also added a spec to avoid this bug in the future.